File: CashuWallet/Models/WalletSupport/RestoreMintResult.swift
import Foundation

struct RestoreMintResult: Identifiable {
    var id: String { mintUrl }
    let mintUrl: String
    let mintName: String
    let spent: UInt64
    let unspent: UInt64
    let pending: UInt64

    var totalRecovered: UInt64 { unspent + pending }
}

/// Token parsed information

File: android/app/src/main/java/org/cashu/wallet/Models/WalletSupport/RestoreMintResult.kt
package org.cashu.wallet.Models

import kotlinx.serialization.Serializable

@Serializable
data class RestoreMintResult(
    val mintUrl: String,
    val mintName: String,
    val spent: Long,
    val unspent: Long,
    val pending: Long,
) {
    val id: String get() = mintUrl
    val totalRecovered: Long get() = unspent + pending
}
